### PR TITLE
Remove trivial predicate for matmul

### PR DESCRIPTION
**Pull request by zasdfgbnm**
This PR is stacked on https://github.com/NVIDIA/Fuser/pull/105, it fixes https://github.com/NVIDIA/Fuser/issues/95.

There are mainly two changes effective:
- In `lower_scalar_hoist.cpp`, add assumptions about parallel indices
- In `predicate_compute.cpp`, check if extent is the same as parallel dimension

Only one of the above changes is sufficient to fix the bug, but I did both to be double safe.

cc: @mmigdal-nv @drzejan2

### CODE REVIEW

File: csrc/predicate_compute.cpp [hunk 1]
```diff
@@ -61,6 +61,14 @@ Bool* ParallelizedDomainPredicate::PredicateInfo::getPredicate() const {
         GpuLower::current()->caMap()->getConcreteMappedID(
             pred_id, IdMappingMode::EXACT));
     auto new_pred = SimplifyingIrBuilder::ltExpr(index, pred_id->extent());
+    // pt_ not being exact does not mean the predicate is not trivial. For
```
Comment (naoyam) [inline csrc/predicate_compute.cpp:64]: This makes sense, but wouldn't it be nicer if such IDs were not added to the predicate ID list from the beginning? Can this filtering be done at around line 211?

Comment (zasdfgbnm) [reply csrc/predicate_compute.cpp:64]: I agree with what you said. Let me give it a try.

Comment (zasdfgbnm) [reply csrc/predicate_compute.cpp]: fixed, it's working

**Review by naoyam — APPROVED** [review]
LGTM

**Pull request merged**